### DISCUSSION TITLE
making one filter disappear when another is active

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1264] [General] [answered]
I have two filters in my CRUD:

```
        CRUD::filter('unpaid')
            ->type('simple')
            ->whenActive(function() {
                CRUD::addClause('unpaid');
            })->else(function() {
                CRUD::addClause('paid');
            });
        CRUD::filter('paid_online')
            ->type('simple')
            ->whenActive(function() {
                CRUD::addClause('paidOnline');
                //CRUD::filter('unpaid')->remove();
            });
```

Here are the relevant methods in the model that correspond to the addClause calls:

```
    public function scopePaid(Builder $query): void
    {
        $query->whereNotNull('datePaid');
    }

    public function scopeUnpaid(Builder $query): void
    {
        $query->whereNull('datePaid');
    }

    public function scopePaidOnline(Builder $query): void
    {
        $query->whereNotNull('paypalID');
    }
```
The thing is...  it doesn't make sense to even show a filter for "paid online" when the "unpaid" filter is active. Like if you've paid online you've implicitly paid. By that same logic it also doesn't make sense to show the "paid online" filter when the "unpaid" filter is active.

Any ideas as to how I can make one filter disappear when another is active? I tried adding `CRUD::filter('unpaid')->remove();` to the `whenActive()` anonymous function but that didn't do the trick.

[a comment by an automated account is not shown]

**Comment by karandatwani92** [comment, marked as answer]
Hey @prescriptionlifeline 

The filters works on ajax and they can't be removed without page reload. 

For now, you can use [dropdown](https://backpackforlaravel.com/docs/6.x/crud-filters#dropdown-1) or [select2](https://backpackforlaravel.com/docs/6.x/crud-filters#select2-1) filter in your use case, so one status is selected at a time.
For example:
```php
CRUD::filter('status')
    ->type('dropdown')
    ->values([
        'unpaid' => 'unpaid',
        'paidOnline' => 'paidOnline',
    ])
    ->whenActive(function ($value) {
        CRUD::addClause('where', 'status', $value);
    });
```
